compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.desktop.kt

```diff
@@ -142,6 +141,7 @@ internal class ComposeSceneMediator(
 
     private val platformComponent = DesktopPlatformComponent()
     private val textInputService = DesktopTextInputService(platformComponent)
```

Comment:
Since the usages of startInput/stopInput were removed, I'm not sure if this should be still here.

If it should, can we have some facade to deal with all versions from one place?

Reply:
> it's still used by TextField(TextFieldValue)

Will it work after removing startInput/stopInput/notifyFocusedRect/updateState calls?

> what you mean

Since this PR seems to drop some calls for one version and I'm not sure it's valid the idea for "facade" is to have some separate class that can encapsulate dealing with all service versions. Like
```
internal class DesktopTextInputService : PlatformTextInputService, PlatformTextInputService2 {
    ....
}
```
It's more like nitpick, so up to you here

Reply:
> Will it work after removing startInput/stopInput/notifyFocusedRect/updateState calls?

Why would they be removed? They're called (on the old `PlatformTextInputService`) by `TextFieldDelegate` and there's no plan to remove that.

We can only remove `PlatformTextInputService` after all of our platforms (iOS, web) implement `PlatformTextInputService2` by delegating to it in `legacyTextInputServiceAdapterAndService()`. And even that's uncertain, because we'd be breaking backwards compatibility for developers who implemented their own `PlatformTextInputService`.

Reply:
> ```
> internal class DesktopTextInputService : PlatformTextInputService, PlatformTextInputService2 {
>     ....
> }
> ```

It's an interesting idea but I think actually trying to implement it will run into issues solving which will make the code more confusing, and will negate the benefit of the facade. Namely, keeping track of which text field version is currently "active" and working accordingly.

Reply:
> Why would they be removed? 

I see the removal of these calls from this mediator (lines 789, 794, 802, 810). I don't see where it's called now. Could you please point to the specific place?

Reply:
The calls that were removed were the ones made for BTF2. The calls for the old text field are made from `compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/TextFieldDelegate.kt`. For example: https://github.com/JetBrains/compose-multiplatform-core/blob/f645d80afe6f2a089238b06b10a1ef496ff33248/compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/TextFieldDelegate.kt#L339